File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            
            if (ch != ' ') { 
                char base = Character.isLowerCase(ch) ? 'a' : 'A'; 
                char temp = (char) (base + (ch - base + n) % 26); 
                answer += temp;
            } else {
                answer += ' '; 
            }
        }
        return answer;
    }
}

    

File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    var answer = '';
    for(let i=0; i<s.length; i++){
        if(s.charAt(i) == ' '){
            answer+=' ';
        }else if(s.charAt(i) >='A' && s.charAt(i) <='Z'){
            answer += String.fromCharCode((s.charAt(i).charCodeAt(0) - 'A'.charCodeAt(0) + n) % 26 + 'A'.charCodeAt(0));
        }else{
            answer += String.fromCharCode((s.charAt(i).charCodeAt(0) - 'a'.charCodeAt(0) + n) % 26 + 'a'.charCodeAt(0));
        }
    }
    return answer;
}